lib/vagrant-service-manager/action/restart_service.rb

```diff
@@ -0,0 +1,34 @@
+module Vagrant
+  module ServiceManager
+    module Action
+      class RestartService
+        def initialize(app, env)
+          @app = app
+          @machine = env[:machine]
+          @ui = env[:ui]
+          @providers = @machine.config.servicemanager.providers.split(',').map(&:chomp)
+        end
+
+        def execute_command(command, provider)
+           @machine.communicate.execute(command) do |type, data|
+            if type == :stderr
+              @ui.error(data)
+            else
+              @ui.info("# Restarted #{provider} service.")
+            end
+          end
+        end
+
+        def call(env)
+          @providers.each do |provider|
+            command = "sudo systemctl restart #{provider}"
```

Comment:
@bexelbie 

> worry about someone passing in unexpected values. i.e. vagrant service-manager restart httpd - if we dont' have apache defined.
Since the `config + hook` code will executed on `vagrant up` which is based on Vagrantfile configuration for `providers(config.servicemanager.providers)`, we need not to worry about someone passing unexpected values. 
Moreover, we can also validate the providers list passed in `Vagrantfile`.